packages/code-editor/src/CodeEditor/stories/CustomPlugin/CustomPlugin.tsx

```diff
@@ -0,0 +1,78 @@
+import { useRef, useState } from "react";
+import { Monaco } from "@monaco-editor/react";
+import {
+  HvCodeEditor,
+  HvCodeEditorProps,
+  hvLanguagePlugins,
+  hvXmlFormatter,
+} from "@hitachivantara/uikit-react-code-editor";
+
+import { Header } from "./utils";
+
+const completionProvider = (monaco: Monaco) => {
+  return {
+    provideCompletionItems: () => {
+      const suggestions = ["Suggestion1", "Suggestion2", "Suggestion3"].map(
+        (label) => ({
+          label,
+          kind: monaco.languages.CompletionItemKind.Keyword,
+          insertText: label,
+          documentation: `Custom suggestion: ${label}`,
+          sortText: "1",
+        }),
+      );
+      return {
+        suggestions,
+      };
+    },
+  };
+};
+
+const customPlugin = {
+  ...hvLanguagePlugins.xml,
+  completionProvider,
+};
+
+export const CustomPluginStory = () => {
+  const [editorValue, setEditorValue] = useState("");
+
+  const editorRef = useRef<any>(null);
+  const monacoRef = useRef<any>(null);
+
+  const handleMount: HvCodeEditorProps["onMount"] = (editor, monaco) => {
+    editorRef.current = editor;
+    monacoRef.current = monaco;
+  };
+
+  const handleFormat = async () => {
+    try {
+      const content = editorRef.current.getValue();
+      const formattedCode = await hvXmlFormatter(
```

Comment:
(just a thought for later)

instead of exporting the plugin's internals, users could use these utils via the plugin itself

```suggestion
      const formattedCode = await xmlPlugin.formatter(
```